### ISSUE TITLE
customer segment issue after data migration magento1 to magento2 enterprise edition 

### ISSUE THREAD

**Issue by popatkaran** [magento/magento2#7203]
I have completed with data migration and data was migrated from magento 1 (ee : 1.12.0.2) to magento 2 (ee: 2.1.1). After migration I have reindex data and cleared cache. In front-end I am getting error

```
a:4:{i:0;s:470371:"SQLSTATE[42S22]: Column not found: 1054 Unknown column 'main.entity_id' in 'field list', query was: SELECT `main`.`entity_id` FROM `catalog_product_entity_int` AS `main`
 INNER JOIN `store` ON main.store_id=store.store_id WHERE (main.attribute_id = '175') AND (store.website_id IN(0, '1')) AND (main.value = '85') AND (main.entity_id IN('2383', '1126', ..... 
```

I have attached report log file here for reference. 
[1272167402754.zip](https://github.com/magento/magento2/files/555056/1272167402754.zip)

If I disable **Magento_CustomerSegment** extension my website is working fine but it must work properly including this extension.

Any help is appreciated.

**Comment by victor-v-rad** [comment]
Hi @popatkaran 

There is a dedicated [place](https://github.com/magento/data-migration-tool/issues) for the Data Migration Tool issues. Please report there. 
As for your error. For quick fix try to change SQL query in Magento 2 DB -> enterprise_customersegment_segment.condition_sql field. I believe replacing 'entity_id' to 'row_id' will fix your issue. But in the migration tool the root of problem should be hidden in \Migration\Handler\Rule\CustomerSegmentConditionSql

**Comment by victor-v-rad** [comment]
Internal issue MAGETWO-60207 for the tool to fix it

**Comment by popatkaran** [comment]
Hi @victor-v-rad 

Thanks for suggestion. I have changed "entity_id" to "row_id" in following two files and it solved my issue in frontend also but I am not sure I that if I am on proper way or not. I have updated 

> /vendor/magento/module-customer-segment/Model/Segment/Condition/Product/Attributes.php
> /vendor/magento/module-customer-segment/Model/Segment/Condition/Product/Combine.php

files. Still I am struggling in admin area for adding new segment and edit existing one. Any help regarding this will be grateful.

**Comment by magento-engcom-team** [comment]
@popatkaran, thank you for your report.
The issue is already fixed in 2.2.0
